src/go/sysfail.go

```diff
@@ -0,0 +1,186 @@
+package sysfail
```

Comment:
Neat! We should perhaps switch FFI test to use this (in a separate PR, at some point).